misc/testerina/modules/testerina-core/src/test/java/org/ballerinalang/testerina/test/DependsOnTest.java

```diff
@@ -56,7 +59,23 @@ public void tesDependsOnFunctionsWithBefore() {
         Assert.assertEquals(runner.getTesterinaReport().getTestSummary(".", "failed"), 0);
     }
 
+    @Test
+    public void tesDependsOnFunctionsMissingFunction() {
+
+        cleanup();
+        BTestRunner runner = new BTestRunner();
+        try {
+            runner.runTest(sourceRoot, new Path[]{Paths.get("depends-on-negative.bal")}, new
+                ArrayList<>());
+
+        } catch (Throwable e) {
+            Assert.assertTrue(e instanceof BallerinaException);
+            Assert.assertTrue(e.getMessage().contains("Cannot find the specified dependsOn function : non-existing"));
```

Comment:
This could be done  at the annotation level itself;
eg:
`@Test(expectedExceptions = BallerinaException.class, expectedExceptionsMessageRegExp = ".*Data provider " +
            "function \\[invalidDataGen\\] should return an array of arrays.*")`